modules/lang-painless/src/main/java/org/elasticsearch/painless/FunctionRef.java

```diff
@@ -97,10 +100,13 @@ public FunctionRef(Class<?> expected, Method interfaceMethod, Method delegateMet
         // the Painless$Script class can be inferred if owner is null
         if (delegateMethod.owner == null) {
             delegateClassName = CLASS_NAME;
+            isDelegateInterface = 0;
         } else if (delegateMethod.augmentation != null) {
             delegateClassName = delegateMethod.augmentation.getName();
+            isDelegateInterface = delegateMethod.augmentation.isInterface() ? 1 : 0;
```

Comment:
This'd only come up if the target augmentation method is defined on an interface, right? Maybe we should not allow that at all.

Reply:
Agreed, but I'd like to do that in a follow up PR as we discussed.